File: back/src/main/java/com/back/domain/report/report/dto/ReportResponseDto.java
package com.back.domain.report.report.dto;

import com.back.domain.report.report.entity.Report;
import com.back.domain.report.report.entity.ReportStatus;
import java.time.LocalDateTime;
import java.util.UUID;

public record ReportResponseDto(
        UUID reportId,
        ReportStatus status,
        LocalDateTime createdAt
) {
    public ReportResponseDto(Report report) {
        this(
                report.getUuid(),
                report.getStatus(),
                report.getCreatedAt()
        );
    }
}

File: back/src/main/kotlin/com/back/domain/report/report/dto/ReportResponseDto.kt
package com.back.domain.report.report.dto

import com.back.domain.report.report.entity.Report
import com.back.domain.report.report.entity.ReportStatus
import java.time.LocalDateTime
import java.util.UUID

data class ReportResponseDto(
    val reportId: UUID,
    val status: ReportStatus,
    val createdAt: LocalDateTime
) {
    constructor(report: Report) : this(report.uuid, report.status, report.createdAt!!)
}
